pkg/testutil/env/sandbox.go

```diff
@@ -153,19 +155,71 @@ func createNamespace(s Sandbox) error {
 		return errors.Wrapf(err, "failed to create namespace: %s", s.Namespace)
 	}
 
+	if err := kube.CreateDatabaseServiceAccount(context.TODO(), s.env.Clientset, s.Namespace); err != nil {
+		return errors.Wrapf(err, "failed to create service account: %s", s.Namespace)
+	}
 	return nil
 }
 
 func createServiceAccount(s Sandbox) error {
-	sa := &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: s.Namespace,
-			Name:      "cockroach-database-sa",
+
+	name := "cockroach-database-sa"
+
+	if _, err := s.env.Clientset.CoreV1().ServiceAccounts(s.Namespace).Create(
+		context.TODO(),
+		&v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: s.Namespace,
+				Name:      name,
+			},
 		},
+		metav1.CreateOptions{},
+	); err != nil {
+		return err
 	}
 
-	if _, err := s.env.Clientset.CoreV1().ServiceAccounts(s.Namespace).Create(context.TODO(), sa, metav1.CreateOptions{}); err != nil {
-		return errors.Wrapf(err, "failed to create service account cockroach-database-sa in namespace %s", s.Namespace)
+	if _, err := s.env.Clientset.RbacV1().ClusterRoleBindings().Create(
+		context.TODO(),
+		&rbac.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cockroach-database-rolebinding",
+			},
+			RoleRef: rbac.RoleRef{
+				Name:     "cockroach-database-role",
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+			},
+			Subjects: []rbac.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      name,
+					Namespace: s.Namespace,
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	); err != nil {
+		return err
+	}
+
+	if _, err := s.env.Clientset.RbacV1().ClusterRoles().Create(
+		context.TODO(),
+		&rbac.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cockroach-database-role",
+			},
+			Rules: []rbac.PolicyRule{
+				{
+					Verbs:         []string{"use"},
+					APIGroups:     []string{"security.openshift.io"},
```

Comment:
Naive question, what does this do when running on GKE? Is it a `no-op`?

Reply:
All of these e2e changes I need to back out and put into another PR.  I have the e2e passing on openshift for one of the tests, so it is passing and I will just do a different PR with the sts changes.